File: ItMe.Client/wwwroot/interop.js
"use strict";

window.interop = {
    toggleClass(element, className) {
        if (window.interop.disableClass(element, className)) {
            element.classList.remove(className);
            return false;
        }
        else {
            return window.interop.enableClass(element, className);
        }
    },

    setDocumentTitle(title) {
        document.title = title;
    },

    setWindowLocation(url) {
        window.location = url;
    }
};

File: ItMe.Client/Utils/Interop.cs
using System.Threading.Tasks;
using Microsoft.AspNetCore.Blazor;
using Microsoft.JSInterop;

namespace ItMe.Client.Utils
{
    public static class Interop
    {
        public static async Task<bool> ToggleClass(ElementRef element, string className)
        {
            return await JSRuntime.Current.InvokeAsync<bool>("interop.toggleClass", element, className);
        }

        public static async Task SetDocumentTitle(string title)
        {
            await JSRuntime.Current.InvokeAsync<object>("interop.setDocumentTitle", title);
        }

        public static async Task SetWindowLocation(string url)
        {
            await JSRuntime.Current.InvokeAsync<object>("interop.setWindowLocation", url);
        }
    }
}